launcher/internal/signaturediscovery/client_test.go

```diff
@@ -48,17 +49,55 @@ func TestFormatSigTag(t *testing.T) {
 
 func TestFetchSignedImageManifestDockerPublic(t *testing.T) {
 	ctx := namespaces.WithNamespace(context.Background(), "test")
-	containerdClient, err := containerd.New(defaults.DefaultAddress)
-	if err != nil {
-		t.Skipf("test needs containerd daemon: %v", err)
-	}
-	defer containerdClient.Close()
 
 	targetRepository := "gcr.io/distroless/static"
 	originalImageDesc := v1.Descriptor{Digest: "sha256:9ecc53c269509f63c69a266168e4a687c7eb8c0cfd753bd8bfcaa4f58a90876f"}
-	client := New(containerdClient, originalImageDesc)
+	client := createTestClient(t, originalImageDesc)
 	// testing image manifest fetching using a public docker repo url
 	if _, err := client.FetchSignedImageManifest(ctx, targetRepository); err != nil {
 		t.Errorf("failed to fetch signed image manifest from targetRepository [%s]: %v", targetRepository, err)
 	}
 }
+
+func TestFetchImageSignaturesDockerPublic(t *testing.T) {
+	ctx := namespaces.WithNamespace(context.Background(), "test")
+	originalImageDesc := v1.Descriptor{Digest: "sha256:905a0f3b3d6d0fb37bfa448b9e78f833b73f0b19fc97fed821a09cf49e255df1"}
```

Comment:
Does the codelab project make this repo public? Otherwise, the test will fail for someone without permissions.

Reply:
Yes, this repo only grants roles/artifactregistry.reader permission to all users, so it is a public accessible repo.